components/layout/fragment.rs

```diff
@@ -1051,7 +1051,8 @@ impl Fragment {
                     // Note: We can not precompute the ratio and store it as a float, because
                     // doing so may result one pixel difference in calculation for certain
                     // images, thus make some tests fail.
-                    inline_size * intrinsic_block_size.0 / intrinsic_inline_size.0
```

Comment:
Better, make a new `AspectRatio` type with a num and a denum, then define operations on it. We may actually utilize the second chapter of SICP :) . But I keeps thinking that this is kind of redundant.